libs/@blockprotocol/graph/src/stdlib/subgraph/element/entity.ts

```diff
@@ -54,80 +66,94 @@ export const getEntity = (
 
-  let entityRevision: Entity | undefined;
-
   // Short circuit for efficiency, just take the latest
   if (targetRevisionInformation === undefined) {
-    [entityRevision] = revisionVersions
+    return revisionVersions
       .slice(-1)
-      .map((latestVersion) => entityRevisions[latestVersion]!.inner);
+      .map((latestVersion) => entityRevisions[latestVersion]!.inner)
+      .pop();
   } else {
     const targetTime =
       typeof targetRevisionInformation === "string"
         ? targetRevisionInformation
         : targetRevisionInformation.toISOString();
 
-    let targetVersion: EntityRevisionId | undefined;
-    for (let idx = 0; idx < revisionVersions.length; idx++) {
-      /** @todo - If we expose endTimes we can do an interval check here per revision, rather than needing to infer it */
-      // Rolling window: we've sorted, so for each revision's version check if the given timestamp is in between the
-      // start of this revision and the next one (lower-bound-inclusive)
-      const [revisionVersion, nextRevisionVersion] = revisionVersions.slice(
-        idx,
-        idx + 1,
-      ) as [EntityRevisionId, EntityRevisionId | undefined];
-
-      // The list is sorted so if this is beyond the target time then all of the ones after are too
-      if (revisionVersion > targetTime) {
-        break;
-      }
+    return revisionVersions
+      .filter((revisionTimestamp) => revisionTimestamp < targetTime)
+      .map((revisionTimestamp) => {
+        const vertex = mustBeDefined(entityRevisions[revisionTimestamp]);
 
-      // Last element in the array (latest version), so we assume an half-closed interval (unbounded on the upper-bound)
-      if (nextRevisionVersion === undefined) {
-        if (revisionVersion <= targetTime) {
-          if (isEntityVertex(entityRevisions[revisionVersion]!)) {
-            targetVersion = revisionVersion;
-          }
-        }
-        break;
-      } else if (
-        revisionVersion <= targetTime &&
-        targetTime < nextRevisionVersion
-      ) {
-        if (isEntityVertex(entityRevisions[revisionVersion]!)) {
-          targetVersion = revisionVersion;
-          break;
+        if (!isEntityVertex(vertex)) {
+          throw new Error(
+            `Found non-entity vertex associated with EntityId: ${entityId}`,
+          );
         }
-      }
-    }
-
-    entityRevision =
-      targetVersion !== undefined
-        ? entityRevisions[targetVersion]!.inner
-        : undefined;
+        return vertex.inner;
+      })
+      .find((entity) => {
+        return intervalContainsTimestamp(
+          /*
+           these casts are safe as we check for `targetRevisionInformation === undefined` above and that's only ever
+           defined if `Temporal extends true`
+           */
+          (entity as Entity<true>).metadata.temporalVersioning[
+            (subgraph as Subgraph<true>).temporalAxes.resolved.variable.axis
+          ],
+          targetTime,
+        );
```

Comment:
Technically there's nothing to stop JS users passing a non-temporal subgraph and `targetRevisionInformation` but I know that worrying about that is a whole can of worms 😁

Reply:
I've been slightly concerned about this too.. I hope they'll at least read the documentation? 😅